core/src/main/scala/shapeless/lazy.scala

```diff
@@ -164,9 +164,34 @@ object Strict {
 }
 
 @macrocompat.bundle
-class LazyMacros(val c: whitebox.Context) extends CaseClassMacros {
+trait OpenImplicitMacros extends compat.CompatLite {
+  val c: whitebox.Context
+
+  import c.universe._
+
+  def openImplicitTpe: Option[Type] =
+    c.openImplicits.headOption.map(x => (x: ImplicitCandidate211).pt)
```

Comment:
Given the implicit conversion https://github.com/milessabin/macro-compat/blob/7297f3152892ea9027fb45fdb2798710f52a7859/core/src/main/scala_2.10/macrocompat/compatcontext.scala#L390

you should be able to just call `x.pt` here.